### ISSUE TITLE
[Newlib/libgloss] Add support for EM SDP board

### ISSUE THREAD

**Issue by abrodkin** [foss-for-synopsys-dwc-arc-processors/toolchain#444]
DesignWare ARC EM Software Development Platform (EM SDP, https://www.synopsys.com/dw/ipdir.php?ds=arc-em-software-development-platform) is the latest and greatest FPGA-based development platform for software development on recent ARC EM processors. It is a more recent addition and to some extent a successor of much older EM Starter Kit board (EMSK, https://www.synopsys.com/dw/ipdir.php?ds=arc_em_starter_kit).
And even though there're similarities between these 2 boards, their memory regions & peripherals are mapped differently, so the same one binary couldn't be re-used. Thus we need to add support of EM SDP as yet another platform in addition to currently available ones.

Basically we need to add:
1. GCC spec: `emsdp.specs`
2. GNU ld linker script: `iotdk.x`
3. UART glue layer: `emsdp-uart-setup.c`

**Comment by kolerov** [comment]
Support of EM SDP is already implemented in Newlib 2023.09. There are 4 `specs` files:

1. `emsdp1.1.specs` - for EM SDP 1.0 and 1.1. Everything in ICCM and DCCM.
2. `emsdp1.1_ram.specs` - for EM SDP 1.0 and 1.1. Everything in RAM, `ivt` in ICCM (for large binaries).
3. `emsdp1.2.specs` - for EM SDP 1.2. Everything in ICCM and DCCM.
4. `emsdp1.2_ram.specs` - for EM SDP 1.2. Everything in RAM, `ivt` in ICCM (for large binaries).
